docs/verification-integration.md

```diff
@@ -0,0 +1,92 @@
+# Verification system
+
+Applications are executed in our Verification system to ensure that:
+
+1. Installation succeeds: All resources are applied and waited for to become healthy.
+1. Functionality tests pass: The deployer starts the Tester Pod and watches its exit status. Zero means
+success, non-zero means failure.
+1. Uninstallation succeeds: Application and all its resources are successfully removed from the cluster.
+
+Successful results are required before application can be published to the GCP Marketplace.
+
+## Functionality tests
+
+Packaging functionality tests involves the following steps.
+
+### 1. Create a `/data-test` folder
+
+The structure is similar to the one in `/data` folder, but with the configuration and manifests that
+are only meant to be applied when the application is installed in verification mode. Some examples of
+`/data-test` for different deployers:
+
+* Helm based deployer:
+
+    ```
+    data-test/
+      chart/
+        nginx.tar.gz
+      schema.yaml
+    ```
+
+* Kubectl based deployer:
+
+    ```
+    data-test/
+      manifest/
+        tester.yaml.template
+      schema.yaml
+    ```
+
+### 2. Include a Pod manifest to execute the tests
+
+The deployer starts the Tester Pod after all the Kubernetes resources are healthy. It should run all
+tests that assert the functionality of the application. If all tests pass, the Tester Pod must complete
+with a zero exit status. Otherwise, it must complete with a non-zero exit status.
+
+The Tester Pod needs to be annotated, so the Verification can detect which Pod to watch for test results.
+The annotation to use depends on the type of the deployer, as follows:
+
+* Helm based deployer: `helm.sh/hook:test-success`
+* Kubectl based deployer: `marketplace.cloud.google.com/verification:test`
```

Comment:
I think this annotation applies to both helm and non-helm. The other annotation should be *in addition* to this when the deployer is helm-based.